### PR TITLE
services: Update candlepin installation 

### PR DESCRIPTION
**Pull request by marusak**
- Don't use Centos 7, but move to current Fedora
- Install rvm.ruby role

 * [x] image-refresh services
 * Needs to land in lockstep with https://github.com/cockpit-project/cockpit/pull/17195

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on 1-cockpit-9.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220328-095858/

File: images/scripts/services.setup [hunk 1]
```diff
@@ -116,27 +116,21 @@ chmod 755 /root/run-samba-domain
 #
 #############################
 
-# unfortunately the setup only works on RHEL/CentOS 7: https://github.com/candlepin/ansible-role-candlepin/pull/12
-# so we have to run that in a CentOS 7 container
-
-podman run --name=candlepin --uts=host --publish=8443:8443 --privileged --detach quay.io/centos/centos:centos7 /sbin/init
+podman run --name=candlepin --uts=host --publish=8443:8443 --privileged --detach -t quay.io/fedora/fedora:35 sh
```
Comment (martinpitt) [inline images/scripts/services.setup:119]: How about `:latest` instead of `:35` so that this keeps itself up to date?

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220328-095858/

**Comment by marusak** [comment]
Was afraid: 
> The conditional check ''firewalld.service' in ansible_facts.services' failed. The error was: error while evaluating conditional ('firewalld.service' in ansible_facts.services): 'dict object' has no attribute 'services'

Not sure why these fact are missing, they should be present

**Comment by marusak** [comment]
These facts [are gathered](https://github.com/candlepin/ansible-role-candlepin/blob/master/tasks/main.yml#L2):
```
TASK [Gathering Facts] *********************************************************
ok: [localhost]
```

**Comment by marusak** [comment]
Oh right, so it gathers facts, but there is no systemd in fedora container so it has no services to collect. That is easy fix. But I wonder if candelpin needs systemd in general or not - lets see.

**Comment by marusak** [comment]
`System has not been booted with systemd as init system (PID 1). Can't operate.`
Ok, it just wants systemd. Lets give it container with it :man_shrugging:

File: images/scripts/services.setup (hunk 1 above)
Comment (marusak) [reply images/scripts/services.setup]: Done

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-d2pvn.
Log: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-3153-20220329-081616/

**Comment by cockpituous** [comment]
Task failed: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-3153-20220329-081616/

**Comment by marusak** [comment]
```
+ podman cp candlepin:/home/candlepin/candlepin/generated_certs /home/admin/candlepin/
Error: "/home/candlepin/candlepin/generated_certs" could not be found on container candlepin: No such file or directory
```

almost there! My script for debugging did not get to these lines :) But the container is working :tada:

**Comment by cockpituous** [comment]
image-refresh in progress on 1-cockpit-8.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220329-092727/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220329-092727/

**Comment by cockpituous** [comment]
Task failed: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-3153-20220329-100000/

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-lm62j.
Log: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-3153-20220329-105454/

**Comment by cockpituous** [comment]
Task failed: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-3153-20220329-105454/

**Comment by marusak** [comment]
> image-upload: unable to upload image: https://cockpit-images.us-east-1.linodeobjects.com/services-601e10d7f24a9002271fb1256096f2ad2ce7491801203909d5912315a228cc3c.qcow2

:sighs:

Today internet is not good - last 3 failures are just networking outages

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-04.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220329-114444/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220329-114444/

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220329-130101/

**Comment by cockpituous** [comment]
image-refresh services done: https://github.com/cockpit-project/bots/commits/image-refresh-services-20220329-135050

**Comment by marusak** [comment]
`wait_js_cond(ph_is_present("[aria-label='Data source plugin item PCP Redis']")): Uncaught (in promise) Error: condition did not become true`

I am having deja vu: https://github.com/cockpit-project/bots/pull/2811/commits/a758ddb6f906fca33bf4e212afe7d2b8978ef489

**Comment by marusak** [comment]
I sent https://github.com/cockpit-project/cockpit/pull/17195 to adjust to the new grafana

**Comment by martinpitt** [comment]
scp: /home/admin/candlepin/generated_certs/6050.pem: No such file or directory

I'm afraid that's a regression (or at least change) from the new candlepin?

**Comment by marusak** [comment]
> I'm afraid that's a regression (or at least change) from the new candlepin?

Yeap, sending PRs now
https://github.com/cockpit-project/bots/pull/3153/files#diff-611cb85c27ec1a4c0d81b3c89462089a64860a4817ed005dd5eed1d87134e222R163

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-02.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-073030/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-073030/

**Comment by marusak** [comment]
I hate internet - `attempting to call Galaxy at 'https://galaxy.ansible.com/api/': <urlopen error [Errno -2] Name or service not known>`

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-10.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-075252/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-075252/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-083232/

**Comment by ptoscano** [comment]
Hello o/ sub-man dev here coming from https://github.com/candlepin/subscription-manager/pull/3015

I see in ansible-role-candlepin that there is a variable `candlepin_home`:
https://github.com/candlepin/ansible-role-candlepin/blob/8f3dd3b9d2d30a5fc02d8aa252f52f89f10ebe01/defaults/main.yml#L4

Maybe that could be set in the local deploy to the old location, so there is no change required on sub-man side? Especially that such a change would be needed not only in `main`, but also in other 5 branches...

**Comment by marusak** [comment]
That has been the case even before, but the difference is that it now contains `/devel` - https://github.com/candlepin/ansible-role-candlepin/commit/8e22ed52ff411425aaec05afe2dc00a0fc68c2b4#diff-23e21b6c89468f7534697ad091835b3ee5e0213b37ace489488234e5a9548ec4L10

But I think we might not need to change it at the end :crossed_fingers:

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-101313/

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-9.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-113838/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-101313/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-113838/

**Comment by cockpituous** [comment]
image-refresh in progress on 1-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-121818/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-121818/

**Comment by cockpituous** [comment]
image-refresh in progress on 1-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-131414/

**Comment by martinpitt** [comment]
@marusak: The most recent failure went rather far, but failed image-upload. But this doesn't yet use the retry logic that I landed this morning. I suggest to rebase this, and try again.

I'll also add some hacks to e2e machines to bandaid the DNS troubles.

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-131414/

**Comment by martinpitt** [comment]
The DNS bandaid is in placae now (https://github.com/cockpit-project/cockpituous/pull/475). The [last refresh](https://logs.cockpit-project.org/logs/image-refresh-3153-20220330-131414/) failed in the candlepin playbook, but maybe that too was a victim of some download. Let's retry this.

I force-pushed a rebase.

**Comment by cockpituous** [comment]
image-refresh in progress on 1-cockpit-7.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220331-070000/

**Comment by cockpituous** [comment]
image-refresh services done: https://github.com/cockpit-project/bots/commits/image-refresh-services-20220331-075050

**Comment by martinpitt** [comment]
The Grafana failures are covered by https://github.com/cockpit-project/cockpit/pull/17195 (adding a blocker for this), but [candlepin is still not happy](https://logs.cockpit-project.org/logs/pull-3153-20220331-081224-b090849c-rhel-9-0-cockpit-project-cockpit/log.html#110-2) :cry:

**Comment by marusak** [comment]
In https://github.com/candlepin/candlepin/pull/3239 default behavior was changed to `Simple Content Access`. Current tests expect the previous behavior. @ptoscano will talk to the team about possibility to tests both cases.

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-01.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-122828/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-122828/

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-121212/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-121212/

**Comment by cockpituous** [comment]
image-refresh in progress on 3-cockpit-10.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-125555/

**Comment by cockpituous** [comment]
image-refresh in progress on 3-cockpit-10.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-132020/

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-01.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-142929/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-142929/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-132020/

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-01.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-154141/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-154141/

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-8.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-162929/

**Comment by martinpitt** [comment]
Bah, so all the recent refreshes are due to Linode uploads being broken? They've been broken so much and consistently recently, that's not really funny any more. I'm EOD today, but next time this happens we should try that interactively on an e2e bot and on a centos CI bot, and locally, and check if that's somehow a networking/proxy problem in e2e, or on the Linode side. But "403" rather sounds like some token permission or bandwidth/volume restriction, not a networking problem as such. @allisonkarlitskaya , do you see anything red on the Linode side? Wrt. quotas or so?

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-3153-20220405-152929/

**Comment by cockpituous** [comment]
image-refresh services done: https://github.com/cockpit-project/bots/commits/image-refresh-services-20220405-172727

File: images/scripts/services.setup [hunk 2]
```diff
@@ -115,44 +115,45 @@ chmod 755 /root/run-samba-domain
...
 
 cat > playbookdir/playbook.yml <<- EOP
 - hosts: localhost
-  environment:
-    JAVA_HOME: /usr/lib/jvm/java-11/
   roles:
      - role: candlepin
-       candlepin_deploy_args: "-g -a -f -t"
+       cp_deploy_args: "-g -a -f -t"
+       cp_git_ref: "896cb9d370f417460d9847171dc965478671c000"
```
Comment (marusak) [inline images/scripts/services.setup:151]: @ptoscano It does not seem to help :/ Still fails the same way

**Comment by martinpitt** [comment]
@marusak : PR #3194 just landed -- so if you rebase, the image refreshes/upload failures should be fixed :partying_face: (Thanks @allisonkarlitskaya !)

File: images/scripts/services.setup (hunk 2 above)
Comment (ptoscano) [reply images/scripts/services.setup:151]: ... aaaaaaand guess what, a bug in the ansible playbook: https://github.com/candlepin/ansible-role-candlepin/pull/26

Comment (ptoscano) [reply images/scripts/services.setup:151]: @marusak the aforementioned playbook issue was fixed; also the test data of candlepin was fixed, so we can try again without setting `cp_git_ref`

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-05.
Log: https://logs.cockpit-project.org/logs/image-refresh-3153-20220407-184646/

**Comment by cockpituous** [comment]
image-refresh services done: https://github.com/cockpit-project/bots/commits/image-refresh-services-20220407-193434

**Comment by ptoscano** [comment]
@marusak I just noticed that we (sub-man) neglected to update `IMAGE_REFRESH_TRIGGERS` for the services image; can you please trigger also `rhel-9-0@candlepin/subscription-manager-cockpit`, please?

**Comment by marusak** [comment]
Work :tada: @martinpitt I have re-triggered https://github.com/cockpit-project/cockpit/pull/17195 as well

**Comment by marusak** [comment]
@ptoscano The context is green (I assume Martin triggered it, thanks!). I sent https://github.com/cockpit-project/bots/pull/3219 to enable this automatically.

**Comment by martinpitt** [comment]
@marusak , @ptoscano : Thanks to both of you, nice job! What a yak shaving..

**Review by martinpitt — APPROVED** [review]

**Pull request merged**